mmpose/datasets/pipelines/top_down_transform.py

```diff
@@ -307,14 +316,112 @@ def _generate_target(self, cfg, joints_3d, joints_3d_visible):
 
         return target, target_weight
 
+    def _megvii_generate_target(self, cfg, joints_3d, joints_3d_visible,
+                                kernel):
+        """Generate the target heatmap via "Megvii" approach.
+
+        Args:
+            cfg (dict): data config
+            joints_3d: np.ndarray ([num_joints, 3])
+            joints_3d_visible: np.ndarray ([num_joints, 3])
+            kernel: Kernel of heatmap gaussian
+        Returns:
+            tuple: A tuple containing targets.
+
+            - target: Target heatmaps.
+            - target_weight: (1: visible, 0: invisible)
+        """
+
+        num_joints = cfg['num_joints']
+        image_size = cfg['image_size']
+        heatmap_size = cfg['heatmap_size']
+        heatmaps = np.zeros((num_joints, heatmap_size[1], heatmap_size[0]),
+                            dtype='float32')
+        target_weight = np.zeros((num_joints, 1), dtype=np.float32)
+
+        for i in range(num_joints):
+            heatmap_vis = joints_3d_visible[i, 0]
+            target_weight[i] = heatmap_vis
+            if target_weight[i] < 1:
+                continue
+            target_y = joints_3d[i, 1] * heatmap_size[1] / image_size[1]
+            target_x = joints_3d[i, 0] * heatmap_size[0] / image_size[0]
+
+            if (target_x >= heatmap_size[0] or target_x < 0) \
+                    or (target_y >= heatmap_size[1] or target_y < 0):
+                target_weight[i] = 0
+                continue
+
+            heatmaps[i, int(target_y), int(target_x)] = 1
+            heatmaps[i] = cv2.GaussianBlur(heatmaps[i], kernel, 0)
+            maxi = np.amax(heatmaps[i])
+            if maxi <= 1e-8:
```

Comment:
in which case can `maxi <= 1e-8`?
from your code, before the blur, heatmap is all zero except for one point. isn't the maximum of heatmap after blur determined by kernel size? do you even need to actually take a maximum over it?

Reply:
This part was copied from the official RSN codes. I think they take the maximum because they needs to map the value range of the heatmap back to [0, 1] by dividing the blurred heatmap by maxi. In case of the divisor maxi being 0, they use if maxi <= 1e-8 then continue. 
I'll remove this part since maxi won't be 0 any way.